ui/src/layouts/storage.rs

```diff
@@ -604,3 +735,58 @@ fn update_items_with_mock_data(
     };
     storage_state.set(Some(storage_mock));
 }
+
+fn get_drag_event() -> FileDropEvent {
+    DRAG_EVENT.read().clone()
+}
+
+fn format_item_name(file_name: String) -> String {
+    let mut file_name_formatted = file_name.clone();
+    let item = PathBuf::from(&file_name);
+
+    let file_stem = item
+        .file_stem()
+        .and_then(OsStr::to_str)
+        .map(str::to_string)
+        .unwrap_or_default();
+
+    if file_stem.len() > MAX_LEN_TO_FORMAT_NAME {
+        file_name_formatted = match &file_name.get(0..15) {
+            Some(name_sliced) => format!(
+                "{}...",
+                name_sliced,
+            ),
+            None => file_name.clone(),
+        };
+    }
+    file_name_formatted
+}
```

Comment:
not a blocker, but the built in functions for `Option`  could simplify the calculation of `file_name_formatted`  - mainly `unwrap_or_else`. 

```
fn format_item_name(file_name: String) -> String {
    let item = PathBuf::from(&file_name);

    let file_stem = item
        .file_stem()
        .and_then(OsStr::to_str)
        .map(str::to_string)
        .unwrap_or_default();

    file_name
        .get(0..15)
        .map(|x| x.to_string())
        .map(|x| {
            if file_stem.len() > MAX_LEN_TO_FORMAT_NAME {
                format!("{x}...")
            } else {
                x
            }
        })
        .unwrap_or_else(|| file_name.clone())
}
```